vdirsyncer/cli/__init__.py

```diff
@@ -151,53 +125,56 @@ def sync(ctx, collections, force_delete, max_workers):
     vdirsyncer sync bob/first_collection
     """
     from .tasks import prepare_pair, sync_collection
-    from .utils import WorkerQueue
 
-    wq = WorkerQueue(max_workers)
+    async def main(collections):
+        conn = aiohttp.TCPConnector(limit_per_host=16)
 
-    with wq.join():
         for pair_name, collections in collections:
-            wq.put(
-                functools.partial(
-                    prepare_pair,
-                    pair_name=pair_name,
-                    collections=collections,
-                    config=ctx.config,
+            for collection, config in await prepare_pair(
+                pair_name=pair_name,
+                collections=collections,
+                config=ctx.config,
+            ):
+                await sync_collection(
+                    collection=collection,
+                    general=config,
                     force_delete=force_delete,
-                    callback=sync_collection,
+                    connector=conn,
                 )
-            )
-            wq.spawn_worker()
+
+    loop = asyncio.get_event_loop()
```

Comment:
get_event_loop is deprecated, use asyncio.run

Reply:
Good catch, thanks!